server/rest/src/main/java/org/infinispan/rest/resources/SearchAdminResource.java

```diff
@@ -108,7 +109,7 @@ public Invocations getInvocations() {
          throw new CacheException("NotImplemented");
       } else {
          SearchStatistics searchStatistics = Search.getSearchStatistics(cache);
-         searchStatistics.getQueryStatistics().clear();
+         Security.doAs(restRequest.getSubject(), () -> searchStatistics.getQueryStatistics().clear());
```

Comment:
why is this required? the cache already contains the subject (via `Cache.withSubject()`)

Reply:
Because that doesn't carry across to the SearchStatistics object